stages/org.osbuild.ostree.coreos.aleph

```diff
@@ -0,0 +1,161 @@
+    }
+
+    write_to_file(tree, ".aleph-version.json", aleph_version_data, coreos_compat)
+
+
+def aleph_container(tree, imgref, coreos_compat):
+    # extract the image name from the imgref
+    imgref_list = imgref.split(':')
+    if imgref_list[0] in ["ostree-remote-registry", "ostree-remote-image"]:
+        img_name = ':'.join(imgref_list[2:])
+    elif imgref_list[0] in ["ostree-image-signed", "ostree-unverified-registry"]:
+        img_name = ':'.join(imgref_list[1:])
+
+    img_name = img_name.lstrip('docker://')
+
+    extra_args = []
+    extra_args.append(f"--repo={tree}/ostree/repo")
+    extra_args.append(f"registry:{img_name}")
+
+    container_data_json = ostree.cli("container", "image", "metadata", *extra_args).stdout.rstrip()
+    container_data = json.loads(container_data_json)
+
+    extra_args.append("--config")
+    container_data_config_json = ostree.cli("container", "image", "metadata", *extra_args).stdout.rstrip()
+    container_data_config = json.loads(container_data_config_json)
+
+    aleph_digest = container_data['config']['digest']
+    aleph_ref = f"docker://{imgref}"
+    aleph_version = container_data_config['config']['Labels']['org.opencontainers.image.version']
+    aleph_container_image = container_data_config['config']['Labels']
+
+    aleph_version_data = {
+        "osbuild-version": osbuild.__version__,
+        "ref": aleph_ref,
+        "build": aleph_version,
+        "version": aleph_version,
+        "container-image": {
+            "image-name": img_name,
+            "image-digest": aleph_digest,
+            "image-labels": aleph_container_image
+        }
+    }
+    
+    # the 'ostree.commit' label will be optional in the future so
+    # prevent hard failing if key is not found
+    aleph_ostree_commit = container_data_config['config']['Labels'].get('ostree.commit')
+    if aleph_ostree_commit is not None:
+        aleph_version_data["ostree-commit"] = aleph_ostree_commit
+
+    write_to_file(tree, ".aleph-version.json", aleph_version_data, coreos_compat)
+
+
+def main(tree, options):
+    coreos_compat = options.get("coreos_compat", False)
+    dep = options.get("deployment", None)
+    osname = ""
+    ref = ""
+    serial = None
+    origin = ""
+
+    # if deployment is specified, use it to find the origin file.
+    # otherwise, autodetect the only deployment found in the tree.
+    if dep is not None:
+        osname = dep.get("osname", "")
+        ref = dep.get("ref", "")
+        serial = dep.get("serial", 0)
+    
+    origin = ostree.deployment_path(tree, osname, ref, serial) + ".origin"
+    
+    # example container case: docker://quay.io/fedora/fedora-coreos:stable
+    # example ostree commit case: fedora:fedora/x86_64/coreos/stable
+    with open(origin) as f:
+        s = f.read()
+    imgref = s.split("=")[1].rstrip()
+    if 'docker://' in imgref:
+        aleph_container(tree, imgref, coreos_compat)
+    else:
+        aleph_commit(tree, imgref, coreos_compat)
```

Comment:
Let's change this code slightly so we don't have to pass around `coreos_compat`. I think I see an easy way: 

```suggestion
    if 'docker://' in imgref:
        data = aleph_container(tree, imgref)
    else:
        data = aleph_commit(tree, imgref)

    # write the data out to the file
    with open(os.path.join(tree, filename), "w") as f:
        json.dump(data, f, indent=4, sort_keys=True)
        f.write("\n")

    # create a symlink for backwards compatibility with CoreOS
    if coreos_compat:
      cwd = os.getcwd()
      os.chdir(tree)
      os.symlink(filename, ".coreos-aleph-version.json")
      os.chdir(cwd)

```